src/include/txn_inline.h

```diff
@@ -225,7 +228,24 @@ __txn_next_op(WT_SESSION_IMPL *session, WT_TXN_OP **opp)
 
     op = &txn->mod[txn->mod_count++];
     WT_CLEAR(*op);
-    op->btree = S2BT(session);
+    btree = S2BT(session);
+    op->btree = btree;
+
+    /*
+     * Store the ID of the latest transaction that is making an update. It can be used to determine
+     * if there is an active transaction on the btree. Only try to update the shared value if this
+     * transaction is newer than the last transaction that updated it.
+     */
+    btree_txn_id_prev = btree->max_upd_txn;
+    txn_id = txn->id;
+    WT_ASSERT_ALWAYS(session, txn_id != WT_TXN_ABORTED,
+      "Assert failure: session: %s: txn->id == WT_TXN_ABORTED", session->name);
+    while (WT_TXNID_LT(btree_txn_id_prev, txn_id)) {
+        if (__wt_atomic_cas64(&op->btree->max_upd_txn, btree_txn_id_prev, txn_id))
```

Comment:
You can just assign this into `op`. No need for an atomic operation. A transaction is part of a session and a session is single-threaded. There can be no other user of `op`. That removes the need for the loop and simplifies this.

Reply:
This code came straight from Sulabh. (His original changes https://github.com/wiredtiger/wiredtiger/pull/8839 that were never merged are the first commit here.) Now I have two different opinions from two people that know much more than me. I need to get a third opinion as a tie breaker.

Reply:
@sulabhM do you recall why you wrote this with an atomic CAS? The `op` array is per-txn which is per-session which is per-thread.

Reply:
The btree data structure is shared among sessions. Sessions may race to change this.

Reply:
As Sulabh said, the atomics are needed.
To make it easier to see what is left I am going to resolve some conversations.